core/broker/broker.go

```diff
@@ -135,6 +136,10 @@ func (b *broker) Init(c *component.Component) error {
 	b.Component.SetStatus(component.StatusHealthy)
 	if b.Component.Monitor != nil {
 		b.monitorStream = b.Component.Monitor.NewBrokerStreams(b.Identity.Id, b.AccessToken)
+
+		if len(viper.GetStringMapString("monitor-servers")) > 0 {
```

Comment:
Part of the cleanup refactor would be depending on config system (i.e. viper, cobra, yml rewriting) only in `cmd` or `utils` package and initialize the components in a clean way.

I'd rather not merge known technical debt, so better fix this. It's an easy fix.